nawab.py

```diff
@@ -4,67 +4,100 @@
 import pwd
 import os
 import threading
+import argparse
 
 #TODO: Shouldn't import this here. Need a way to derive this from tg_bot instead.
 from telegram.ext import Updater, CommandHandler, CallbackQueryHandler, ConversationHandler
 
 
-def retrieve_twitter_auth(dirpath, data):
-    tw_bot = twitter_bot.Twitter_Bot(dirpath, data)
+def retrieve_twitter_auth(dirpath, data, level):
+    tw_bot = twitter_bot.Twitter_Bot(dirpath, data, level)
     api = tw_bot.nawab_twitter_authenticate()
     return tw_bot, api
 
 
-def twitter_bot_run(data, dirpath):
-    tw_bot, api = retrieve_twitter_auth(dirpath, data)
+def twitter_bot_run(data, dirpath, level, isretweet=False):
+    tw_bot, api = retrieve_twitter_auth(dirpath, data, level)
     tw_bot.nawab_curate_list(api)
-    tw_bot.nawab_retweet_tweet(api)
+    if isretweet == True:
+        tw_bot.nawab_retweet_tweet(api)
 
 
-def tg_bot_run(data, dirpath):
-    tw_bot, api = retrieve_twitter_auth(dirpath, data)
-    bot = tg_bot.Telegram_Bot(api, dirpath)
+def tg_bot_run(data, dirpath,level, isretweet=False):
+    tw_bot, api = retrieve_twitter_auth(dirpath, data, level)
+    bot = tg_bot.Telegram_Bot(api, dirpath, level)
     updater = bot.nawab_tg_authenticate()
 
     dp = updater.dispatcher
     dp.add_handler(CommandHandler('start', bot.start))
-    dp.add_handler(CommandHandler("stop", bot.stop, pass_chat_data=True))
+    dp.add_handler(CommandHandler('stop', bot.stop, pass_chat_data=True))
     dp.add_handler(CommandHandler('help',bot.help))
     dp.add_handler(CallbackQueryHandler(bot.button))
     dp.add_error_handler(bot.error)
 
     # Start the Bot
-    updater.start_polling()
-
+    if isretweet == False:
+        updater.start_polling()
+    
     # Run the bot until the user presses Ctrl-C or the process receives SIGINT,
     # SIGTERM or SIGABRT
     updater.idle()
 
-
 def wrapper(func, args, res):
     res.append(func(*args))
 
 
 def main():
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument("-r", "--retweet", help="Retweet all tweets automatically, doesn't spawn a telegram bot",
+                        action='store_true', required=False)
+    parser.add_argument("-b", "--blacklist",type=list, required=False, help="Blacklist the given username")
+    parser.add_argument("-p", "--path", type=str, required=False,
+                        help="Path where the log files be stored. Note to create directory in that path beforehand.")
+    parser.add_argument('-d', '--default', action="store_const", const=30)
+    parser.add_argument('-V', '--verbose',action="store_const", const=20)
+    parser.add_argument('-s', '--silent', action="store_const", const=50)
```

Comment:
Does these arguments take some sort of inputs?

Reply:
No. it does not